client/src/templates/Challenges/utils/frame.ts

```diff
@@ -267,9 +267,44 @@ const writeContentToFrame = (frameContext: Context) => {
     createHeader(frameContext.element.id) + frameContext.build,
     frameContext.document
   );
+
+  registerScrollEventListner(frameContext.element);
+
+  if (getPreviewScrollPosition()) {
+    restorePreviewScrollPosition(
+      frameContext.element,
+      getPreviewScrollPosition()
+    );
+  }
   return frameContext;
 };
 
+let previewScrollPosition = 0;
+export const getPreviewScrollPosition = () => previewScrollPosition;
+export const setPreviewScrollPosition = (position: number) =>
+  (previewScrollPosition = position);
+
+const registerScrollEventListner = (iframe: HTMLIFrameElement) => {
+  iframe.contentDocument?.addEventListener(
+    'scroll',
+    (event: IframeEvent<Document>) => {
```

Comment:
```suggestion
    (event) => {
```
This type doesn't exist, but `Event` is fine.